jpamb/cli.py

```diff
@@ -531,28 +562,26 @@ def build(suite, compile, decompile, document, test, docker):
         document = document is None
         test = test is None
 
-    dockerbin = shutil.which("podman") or shutil.which("docker")
-
-    if not dockerbin:
-        raise click.UsageError("No docker or podman on PATH")
-
-    log.info(f"Using docker: {dockerbin}")
-
-    cmd = [
-        dockerbin,
-        "run",
-        "--rm",
-        "-v",
-        f"{suite.workfolder}:/workspace",
-        docker,
-    ]
+    docker_cmd = get_docker_command()
```

Comment:
Looks good! 

Can you refactor this so that it returns a `docker` object with a `run` method instead which we can use to replace the commands below?